Doc/tutorial/errors.rst

```diff
@@ -147,6 +147,36 @@ For example, the following code will print B, C, D in that order::
 Note that if the *except clauses* were reversed (with ``except B`` first), it
 would have printed B, B, B --- the first matching *except clause* is triggered.
 
+When an exception occurs, it may have associated values, also known as the
+exception's *arguments*. The presence and types of the arguments depend on the
+exception type.
+
+The *except clause* may specify a variable after the exception name.  The
+variable is bound to the exception instance which typically has an ``args``
+attribute that stores the arguments. For convenience, builtin exception
+types define :meth:`__str__` to print all the arguments without explicitly
+accessing ``.args``.  ::
+
+   >>> try:
+   ...     raise Exception('spam', 'eggs')
+   ... except Exception as inst:
+   ...     print(type(inst))    # the exception instance
+   ...     print(inst.args)     # arguments stored in .args
+   ...     print(inst)          # __str__ allows args to be printed directly,
+   ...                          # but may be overridden in exception subclasses
+   ...     x, y = inst.args     # unpack args
+   ...     print('x =', x)
+   ...     print('y =', y)
+   ...
+   <class 'Exception'>
+   ('spam', 'eggs')
+   ('spam', 'eggs')
+   x = spam
+   y = eggs
+
+The exception's :meth:`__str__` output is printed as the last part ('detail')
+of the message for unhandled exceptions.
+
 All exceptions inherit from :exc:`BaseException`, and so it can be used to serve
 as a wildcard. Use this with extreme caution, since it is easy to mask a real
```

Comment:
Not a fan of this section!  `except Exception` is the catch-all recommended by style guides and linting tools; you have to know what you’re doing when you write `except BaseException`.

I saw this was discussed in the ticket but didn’t read all messages.  Do you think it’s in scope for this ticket, or should be a discussion somewhere with a wider core dev audience?

Reply:
We can discuss it here I think. I agree it's a dodgy thing to have in a tutorial.  And we probably don't want to talk about the no-type except: clause here either.